pkg/tsdb/influxdb/models/query.go

```
@@ -44,6 +95,96 @@ func (query *Query) Build(queryContext *backend.QueryDataRequest) (string, error
 	return res, nil
 }
 
+func (query *Query) renderAdhocFilters() []string {
+	res := make([]string, 0, len(query.AdhocFilters))
+	for i, tag := range query.AdhocFilters {
+		str := ""
+
+		if i > 0 {
+			if tag.Condition == "" {
+				str += "AND"
+			} else {
+				str += tag.Condition
+			}
+			str += " "
+		}
+
+		// If the operator is missing we fall back to sensible defaults
+		if tag.Operator == "" {
+			if regexpOperatorPattern.MatchString(tag.Value) {
+				tag.Operator = "=~"
+			} else {
+				tag.Operator = "="
+			}
+		}
+
+		isOperatorTypeHandler := func(tag *Tag) (string, string) {
+			// Attempt to identify the type of the supplied value
+			var lowerValue = strings.ToLower(tag.Value)
+			var r = regexp.MustCompile(`^(-?)[0-9\.]+$`)
+			var textValue string
+			var operator string
+
+			// Perform operator replacements
+			switch tag.Operator {
+			case "Is":
+				operator = "="
+			case "Is Not":
+				operator = "!="
+			default:
+				// This should never happen
+				operator = "="
```

If this should never, should we return an error?